design-documents/frontend/admin-guidelines.md

```diff
@@ -0,0 +1,194 @@
+
+For more information behind these technology choices, refer to the [Technical Vision](./technical-vision.md).
+
+## package.json
+
+Each section of the `package.json` file has been broken down into different topics.
+
+- [version](#version)
+- [scripts](#scripts)
+- [browserslist](#browserslist)
+- [dependencies](#dependencies)
+  - [Utility functions](#utility-functions)
+  - [SSR](#ssr)
+  - [React](#react)
+  - [State management](#state-management)
+  - [Code splitting](#code-splitting)
+  - [i18n](#i18n)
+  - [Theming](#theming)
+  - [Relay](#relay)
+  - [TypeScript import helpers](#typescript-import-helpers)
+- [devDependencies](#devdependencies)
+  - [@types](#types)
+  - [Redux mock store](#redux-mock-store)
+  - [Testing library](#testing-library)
+
+### version
+
+The version of this `package.json` follows [semver](https://semver.org/) and will change with this document accordingly.
+
+```json
+"version": "0.1.0",
+```
+
+### scripts
+
+Customize your CRA configuration with [craco][].
+
+- [Don't eject](https://create-react-app.dev/docs/alternatives-to-ejecting).
+- Avoid forking, because you can oftentimes do the same thing by creating your own [craco][] plugin. See [craco plugins on npm](https://www.npmjs.com/search?q=craco%20plugin) and the [craco recipes](https://github.com/sharegate/craco/tree/master/recipes) for guidance.
+
+```json
+"scripts": {
+  "start": "craco start",
+  "build": "craco build",
+  "test": "craco test"
+},
+```
+
+### [browserslist](https://github.com/browserslist/browserslist#readme)
+
+Don't forget your [polyfills](https://create-react-app.dev/docs/supported-browsers-features#supported-language-features)!
+
+```json
+"browserslist": [
+  ">0.2%",
+  "not dead",
+  "not ie <= 11",
+  "not op_mini all"
+],
+```
+
+### dependencies
+
+#### Utility functions
+
+Use vanilla (native) JS if possible, utility functions if not. [You don't (may not) need Lodash/Underscore](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#you-dont-may-not-need-lodashunderscore-). See also [just](https://github.com/angus-c/just#just).
+
+```json
+"@queso/camel-case": "^0",
+"@queso/kebab-case": "^1",
+```
+
+#### SSR
+
+Prepare for [SSR](https://medium.com/walmartlabs/the-benefits-of-server-side-rendering-over-client-side-rendering-5d07ff2cefe8) by using [universal fetching](https://www.npmjs.com/package/ky-universal). Don't use [axios](https://www.npmjs.com/package/axios) or [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) directly.
+
+```json
+"ky": "^0",
+"ky-universal": "^0",
+```
+
+#### React
+
+```json
+"react": "^16",
+"react-dom": "^16",
+"react-helmet": "^5",
+```
+
+#### Routing
+
+See [The Future of React Router and @reach/router](https://reacttraining.com/blog/reach-react-router-future/). Basically, the new API will be much closer to `@reach/router` than React Router, so this prepares us for that future.
+
+```json
+"@reach/router": "^1",
+```
+
+#### State management
```

Comment:
At this point I exclusively use React's `context`. Thoughts?

Reply:
If you can swing it, that's fantastic! It's still a bit of a new concept, however, so I don't want to take away Redux just yet. I'd be interested in reviewing some code you have to see what the pros & cons are at this point.